### PR TITLE
egs2/magicdata_ramc: add ASR recipe for MagicData-RAMC conversational corpus

### PR DESCRIPTION
**Pull request by HartmannPsi**
## What did you change?

Add an ESPnet2 ASR recipe for MagicData-RAMC (Mandarin two-party conversational speech, OpenSLR #123) under `egs2/magicdata_ramc/asr1/`.

---

## Why did you make this change?

MagicData-RAMC is a free 180-hour Mandarin conversational corpus that is currently not covered by ESPnet, despite being a useful complement to the existing read-speech Mandarin recipes (AISHELL, MAGICDATA). Note that this is distinct from the existing `egs2/magicdata` recipe, which covers the unrelated MAGICDATA Mandarin Read Speech corpus (OpenSLR #68).

The recipe adds:

- `local/data.sh` + `local/prepare_data.py` — automated download from OpenSLR #123, parsing of per-conversation annotations into per-utterance segments, configurable duration / text-length filtering, paralinguistic tag stripping, and Kaldi-style data dir emission for train/dev/test.
- A set of configs and run scripts for a Conformer baseline, a Branchformer-24, an E-Branchformer-12 (with speed perturbation + Transformer LM rescoring), plus warm-start variants for both the ASR encoder (from AISHELL E-Branchformer) and the LM (from a magicdata read-speech LM).
- A `README.md` with the full results table and per-configuration analysis.

---

## Is your PR small enough?

43 files, ~2050 lines added.

---

## Additional Context

The recipe has been run end-to-end on 2 × A800-80GB; full numbers and ablation discussion are in the recipe `README.md`. The best configuration (E-Branchformer + SP + scratch LM) reaches dev CER 13.7 / test CER 17.6.

Checklist:

- [x] `local/data.sh` end-to-end (download → Kaldi-style dirs → validation)
- [x] Stages 1–5 (data prep) verified
- [x] Stage 6–7 (LM stats + train) verified
- [x] Stage 10–11 (ASR stats + train) verified across all 6 configurations
- [x] Stage 12–13 (decode + score) verified across all 6 configurations
- [x] Recipe `README.md` with results table + analysis
- [x] Entry added to `egs2/README.md` overview table
- [x] `MAGICDATA_RAMC` placeholder added to `egs2/TEMPLATE/asr1/db.sh`
- [x] Recipe-local `.gitignore` for `pretrained/` (HF-fetched checkpoints not committed)

### CODE REVIEW

**Comment by sw005320** [comment]
@HartmannPsi, could you check Gemini reviews? 
(Some of them are useful and some of them are not)

**Comment by HartmannPsi** [comment]
Hi Prof. @sw005320 , thanks for the heads-up. Yes — I went through all five Gemini suggestions and applied them in commit `668a890ae0`:

- **`data.sh` — drop `--no-check-certificate` on wget**: real fix. I had added it during local debugging and forgot to remove it.
- **TSV / TXT `utf-8-sig` and backslash normalisation in `prepare_data.py`**: the released MagicData-RAMC files don't actually trigger these (no BOM, forward-slash paths), so these three are purely defensive against re-saved-on-Windows copies. The changes are tiny so I kept them in, but happy to revert if you'd rather keep the diff focused.
- **Always strip `SPECIAL_TAGS` in `clean_text`**: this one was a real bug — `KEEP_PATTERN` removes `[` and `]` but keeps the tag letters, so the previous fallback would have silently corrupted transcripts with `LAUGHTER`/`MUSIC`/etc. Fixed by always stripping the tags; I also removed the now-redundant `--filter-special-symbols` flag and added a docstring note about `non_linguistic_symbols` as the proper long-term home for these markers.

CI is fully green on the new HEAD. Let me know if you'd like any of these revisited.

**Comment by sw005320** [comment]
@HartmannPsi, it looks good to me in general, but many top-level recipe scripts are added as executable, including:

- launch_*.sh
- run_branchformer*.sh
- run_e_branchformer*.sh
- run_decode_warmstart_lm.sh
- train_lm_warmstart.sh

For ESPnet recipe style, it’s better to keep `run.sh` as the main executable entrypoint and move these helper/experiment scripts under `local/` (or merge them into run.sh via options).

**Comment by sw005320** [comment]
Also, `launch_*.sh` seems to be for your debugging purposes and can be removed.

**Comment by HartmannPsi** [comment]
Hi Prof. @sw005320 , addressed your feedback in commit `07b5c095fb`:

- Removed all `launch_*.sh`.
- Consolidated the five `run_*.sh` variants into a single `run.sh` defaulting to the recipe's best config (E-Branchformer + SP + Transformer LM). Other rows of the results table are now reachable by overriding `run.sh`'s defaults at the CLI, mirroring the aishell-style; the README's new "Reproducing the other rows of the results table" section lists the exact override for each.
- Moved `train_lm_warmstart.sh` to `local/` since it's a recipe-local helper (asr.sh stage 7 has no `--pretrained_lm` passthrough, so this script invokes `espnet2.bin.lm_train` directly with `--init_param`).

The latest CI run shows 5 failures + a large cluster of cancellations, but they look like infrastructure / flakiness issues rather than problems with the recipe:

- All five failing jobs are in `espnet2`/`espnet3` Python / integration / configuration suites — none exercise code in `egs2/magicdata_ramc/`, the only directory this PR touches.
- They all died in 2–9 minutes (well under the normal 1–2h runtime), suggesting setup / import failures rather than actual test logic.
- Upstream `master` HEAD (`295ef69ae`) is green on those same five job names on the same Python × PyTorch × NumPy matrix.
- The 49 cancellations match the usual concurrency-cancel pattern triggered by my "Merge master into MagicData-RAMC" commit.

Would you mind re-running these from the maintainer side? Happy to dig into any specific failure if you'd like — I just don't see anything in the recipe diff that would touch those test paths.

**Comment by sw005320** [comment]
Sorry, I'm working on fixing the CI issue in https://github.com/espnet/espnet/pull/6455

File: egs2/magicdata_ramc/asr1/README.md [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+a free 180-hour Mandarin **conversational** speech corpus, designed as a harder
+benchmark than the usual read-speech datasets. Each recording is a ~30 min
+two-party spontaneous dialogue, 16 kHz two-channel (one speaker per channel),
+covering 15 everyday and semi-professional topics (daily life, food, sports,
+technology, business, healthcare, etc.). The corpus comes with an official
+train/dev/test split with disjoint speakers and topics.
+
+After the recipe's standard filtering (segments outside 0.3–30 s or text outside
+1–200 chars are dropped, paralinguistic tags such as `[LAUGHTER]`/`[MUSIC]` are
+stripped from the text), the prepared Kaldi-style data directories are:
```
Comment (sw005320) [inline egs2/magicdata_ramc/asr1/README.md]: Do you mean that you performed this filtering for the test sets as well?
We should not apply it to them.

Also, paralinguistic tags can be handled by adding them to nlsyms. You may refer to 
https://github.com/espnet/espnet/blob/master/egs2/wsj/asr1/local/data.sh#L53

File: egs2/magicdata_ramc/asr1/README.md [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+model (12-block E-Branchformer, ~38 M params) from scratch beats both
+warm-start variants. To make warm-start a real win on this corpus would
+require a **conversational-domain** pretrained model — e.g., WenetSpeech-
+class for the encoder (10k+ hours of in-the-wild Mandarin), or an LM
+trained on a real conversational corpus. Those are the only directions
+likely to break the ~13.7 dev / 17.6 test CER plateau set by row 3b.
+
+## TL;DR
+
+Use **row 3b** (`./run_e_branchformer.sh`). On 2 × A800-80GB this takes
```
Comment (sw005320) [inline egs2/magicdata_ramc/asr1/README.md]: I think `run_e_branchformer.sh` should be modified.
Can you check the entire repo to reflect your changes?

**Review by sw005320 — APPROVED** [review]

**Comment by sw005320** [comment]
Thanks!
